Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -390,56 +358,66 @@ - (void)configureViews
-            [bodyTextView autoPinTrailingToSuperviewMarginWithInset:textInsets.trailing],
-            [bodyTextView autoSetDimension:ALDimensionWidth toSize:bodyTextContentSize.width],
             [bodyTextView autoSetDimension:ALDimensionHeight toSize:bodyTextContentSize.height],
         ]];
 
-        if (lastSubview) {
-            [self.viewConstraints addObject:[bodyTextView autoPinEdge:ALEdgeTop
-                                                               toEdge:ALEdgeBottom
-                                                               ofView:lastSubview
-                                                           withOffset:textInsets.top]];
-        } else {
-            [self.viewConstraints
-                addObject:[bodyTextView autoPinEdgeToSuperviewEdge:ALEdgeTop withInset:textInsets.top]];
+        UIView *_Nullable tapForMoreLabel = [self createTapForMoreLabelIfNecessary];
+        if (tapForMoreLabel) {
+            [textStackView addArrangedSubview:tapForMoreLabel];
+            [self.viewConstraints addObjectsFromArray:@[
+                [tapForMoreLabel autoSetDimension:ALDimensionHeight toSize:self.tapForMoreHeight],
+            ]];
         }
-        lastSubview = bodyTextView;
-        bottomMargin = textInsets.bottom;
     }
 
-    UIView *_Nullable tapForMoreLabel = [self createTapForMoreLabelIfNecessary];
-    if (tapForMoreLabel) {
-        OWSAssert(lastSubview);
-        OWSAssert(lastSubview == bodyTextView);
-        [self.bubbleView addSubview:tapForMoreLabel];
+    OWSMessageFooterView *footerView = self.footerView;
+    [footerView configureWithConversationViewItem:self.viewItem];
+    if (textStackView) {
+        [textStackView addArrangedSubview:self.footerView];
+        [self.footerView setHasShadows:NO viewItem:self.viewItem];
+    } else if (bodyMediaView) {
+        [bodyMediaView addSubview:footerView];
+
+        bodyMediaView.layoutMargins = UIEdgeInsetsZero;
         [self.viewConstraints addObjectsFromArray:@[
-            [tapForMoreLabel autoPinLeadingToSuperviewMarginWithInset:textInsets.leading],
-            [tapForMoreLabel autoPinTrailingToSuperviewMarginWithInset:textInsets.trailing],
-            [tapForMoreLabel autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:lastSubview],
-            [tapForMoreLabel autoSetDimension:ALDimensionHeight toSize:self.tapForMoreHeight],
+            [footerView autoPinLeadingToSuperviewMarginWithInset:self.conversationStyle.textInsetHorizontal],
+            [footerView autoPinTrailingToSuperviewMarginWithInset:self.conversationStyle.textInsetHorizontal],
+            [footerView autoPinBottomToSuperviewMarginWithInset:self.conversationStyle.textInsetBottom],
         ]];
-        lastSubview = tapForMoreLabel;
-        bottomMargin = textInsets.bottom;
+        [self.footerView setHasShadows:YES viewItem:self.viewItem];
+    } else {
+        // Display footer over media.
```

Comment:
Is this comment in the right spot? It seems like it belongs in the previous branch, or you could just delete it.